### PR TITLE
fix: update theme object

### PR DESCRIPTION
**Pull request by joaoprcosta**
# Description

Update and refactor the theme object schema.

- new typography schema
- reduce the breakpoints in each schema object

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other:

# Checklist:

- [X] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/react/src/context/utils/utils.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { defaultTheme } from '../../tokens';
```
Comment (joaoprcosta) [inline packages/react/src/context/utils/utils.ts:1]: This file should be renamed to `deepMerge.ts`. Also, the testing file is outside utils folder. Should be inside

File: packages/react/src/context/utils/utils.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    }
+    else {
+      target[key] = sourceValue;
+    }
+  }
+
+  return target;
+};
+
+export const mergeThemes = (newTheme: Partial<Theme>): Theme => {
```
Comment (joaoprcosta) [inline packages/react/src/context/utils/utils.ts:25]: I think we can rename it back to `themes`. as the merge is being done inside.

Comment (joaoprcosta) [inline packages/react/src/context/utils/utils.ts:25]: Also, why is this const inside a util and not at the root of themes?

File: packages/react/src/tokens/themes/theme.types.ts [hunk 3]
```diff
@@ -1,12 +1,59 @@
...
+}
+
+interface TypographyBreakpoints {
+  mobile?: TypographyStyle;
+  tablet?: TypographyStyle;
+  desktop?: TypographyStyle;
+  desktopLarge?: TypographyStyle;
+}
+
+export interface TypographyHeader {
```
Comment (joaoprcosta) [inline packages/react/src/tokens/themes/theme.types.ts:28]: Makes sense for headings and other type of same structures?

```suggestion
export enum Headings {
  H1 = 'h1',
  H2 = 'h2',
  H3 = 'h3',
  H4 = 'h4',
  H5 = 'h5',
  H6 = 'h6'
}

type TypographyHeader = {
  [heading in Headings]: {
    [breakpoint in Breakpoints]: TypographyStyle
  };
}
```

File: packages/react/src/context/utils/utils.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { defaultTheme } from '../../tokens';
+import { Theme } from '../../tokens/themes/theme.types';
+
+const deepMerge = <T>(target: T, source: Partial<T>): T => {
+  const isObject = (obj: any) => obj && typeof obj === 'object';
```
Comment (joaoprcosta) [inline packages/react/src/context/utils/utils.ts:5]: Could this any be 'TypographyBreakpoint' type?

**Review by vzdiogo — APPROVED** [review]

**Review by Arthurmqz — APPROVED** [review]

**Review by JuniorDiasOliveira — APPROVED** [review]

**Pull request merged**